api/app/c_haines/fetch.py

```diff
@@ -0,0 +1,273 @@
+    SeverityEnum.EXTREME: '11+ Extreme'
+}
+
+# Severity enum -> KML style mapping
+severity_style_map = {
+    SeverityEnum.LOW: 'low',
+    SeverityEnum.MODERATE: 'moderate',
+    SeverityEnum.HIGH: 'high',
+    SeverityEnum.EXTREME: 'extreme'
+}
+
+
+def get_severity_style(c_haines_index: SeverityEnum) -> str:
+    """ Based on the index range, return a style string """
+    return severity_style_map[c_haines_index]
+
+
+def open_placemark(model: ModelEnum, severity: SeverityEnum, timestamp: datetime):
+    """ Open kml <Placemark> tag. """
+    kml = []
+    kml.append('<Placemark>')
+
+    if model == ModelEnum.GDPS:
+        end = timestamp + timedelta(hours=3)
+    else:
+        end = timestamp + timedelta(hours=1)
+    kml.append('<TimeSpan>')
+    kml.append('<begin>{}</begin>'.format(timestamp.isoformat()))
+    kml.append('<end>{}</end>'.format(end.isoformat()))
+    kml.append('</TimeSpan>')
+    kml.append('<styleUrl>#{}</styleUrl>'.format(get_severity_style(severity)))
+    kml.append('<name>{}</name>'.format(severity_text_map[severity]))
+    kml.append('<MultiGeometry>')
+    return "\n".join(kml)
+
+
+def close_placemark():
+    """ Close kml </Placemark> tag. """
+    kml = []
+    kml.append('</MultiGeometry>')
+    kml.append('</Placemark>')
+    return "\n".join(kml)
+
+
+def add_style(kml, style_id, color):
+    """ Add kml <Style> tag """
+    kml.append('<Style id="{}">'.format(style_id))
+    kml.append('<LineStyle>')
+    kml.append('<width>1.5</width>')
+    kml.append('<color>{}</color>'.format(color))
+    kml.append('</LineStyle>')
+    kml.append('<PolyStyle>')
+    kml.append('<color>{}</color>'.format(color))
+    kml.append('</PolyStyle>')
+    kml.append('</Style>')
+
+
+async def fetch_prediction_geojson(model: ModelEnum, model_run_timestamp: datetime,
+                                   prediction_timestamp: datetime):
+    """ Fetch prediction polygon geojson.
+    """
+    logger.info('model: %s; model_run: %s, prediction: %s', model, model_run_timestamp, prediction_timestamp)
+    with app.db.database.get_read_session_scope() as session:
+        response = get_prediction_geojson(session, model, model_run_timestamp, prediction_timestamp)
+    return response
+
+
+def get_look_at(model: ModelEnum, model_run_timestamp: datetime):
+    """ Return <LookAt> tag to set default position and timespan.
+    If this isn't done, then all the predictions will show as a big overlayed mess. """
+    if model == ModelEnum.GDPS:
+        end = model_run_timestamp + timedelta(hours=3)
+    else:
+        end = model_run_timestamp + timedelta(hours=1)
+    kml = []
+    kml.append('<LookAt>')
+    kml.append('<gx:TimeSpan>')
+    kml.append(f'<begin>{model_run_timestamp.isoformat()}</begin>')
+    kml.append(f'<end>{end.isoformat()}</end>')
+    kml.append('</gx:TimeSpan>')
+    kml.append('<longitude>-123</longitude>')
+    kml.append('<latitude>54</latitude>')
+    kml.append('<range>3000000</range>')
```

Comment:
(a) what is "range"? (b) why are there hard-coded lat/long values here?

Reply:
a) Distance from point (added a comment) b) fairly arbitrary position that gives you a nice view of B.C. in google earth (added comment).